lib: location: Add support for GCI cell (#11549)
nrfconnect/sdk-nrf: doc/nrf/libraries/modem/location.rst

* Cellular positioning

  * Uses :ref:`lte_lc_readme` for getting a list of nearby cellular base stations.
  * Neighbor cell measurement is performed with :c:enum:`LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_COMPLETE` search type.

>     Then, GCI search with :c:enum:`LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_LIGHT` search type is performed if previous search did not find enough cells.
>     More details on how GCI search is done can be seen from :c:member:`location_cellular_config.cell_count`.

Reviewer: ```suggestion
    A GCI search with :c:enum:`LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_LIGHT` search type is performed if the previous search did not find enough cells.
    For more details on GCI search, see :c:member:`location_cellular_config.cell_count`.
```
